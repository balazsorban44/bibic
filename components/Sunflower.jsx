import React from 'react'
import {Gallery} from './Gallery'

const Sunflower = ({paragraphs, images}) =>
  <section id="napraforgo">
    {paragraphs.map(({text}, index) =>
      <p key={index}>{text}</p>
    )}
    <Gallery images={images}/>
    <div className="cert-bg">
      <img
        aria-hidden
        src="/images/bg/intro.jpg"
      />
    </div>
  </section>

export default Sunflower