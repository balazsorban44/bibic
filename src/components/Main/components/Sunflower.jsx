import React, {Component} from 'react'
import { CERTS_REF } from '../../../lib/firebase';


export default class Sunflower extends Component {

  state = {
    certificates: null
  }

  componentDidMount() {
    CERTS_REF.on("value", snap => {
      this.setState({certificates: snap.val()})
    })
  }

  render() {
    const {certificates} = this.state
    return (
      <section id="napraforgo">
        <p>Vendégházaink <b>négy napraforgós minősítés</b>sel rendelkeznek. Ez azt jelenti, hogy rendelkezünk <b>minden szobá</b>nkhoz külön <b>fürdőszobá</b>val, a házainkban felszerelt <b>konyha</b> és közösségi (<b>nappali</b>) tér van, <b>kerthelyiség</b>ünk, <b>terasz</b>unk bútorozott és zárt <b>parkoló</b>t tartunk fenn vendégeink számára, valamint házunk megfelel a FATOSZ által elvárt minősítési követelményeknek.</p>
        <div className="certificates">
          {certificates  && Object.keys(certificates).map(certificate => {
            return <img key={certificate} alt={certificates[certificate].name} className="certificate" src={certificates[certificate].resized} />
          })}
        </div>
      </section>
    )
  }
}
