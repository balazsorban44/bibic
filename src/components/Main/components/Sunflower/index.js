import React from 'react'

const Sunflower = () => (
  <section id="napraforgo">
    <p>Vendégházaink <b>négy napraforgós minősítés</b>sel rendelkeznek. Ez azt jelenti, hogy rendelkezünk <b>minden szobá</b>nkhoz külön <b>fürdőszobá</b>val, a házainkban felszerelt <b>konyha</b> és közösségi (<b>nappali</b>) tér van, <b>kerthelyiség</b>ünk, <b>terasz</b>unk bútorozott és zárt <b>parkoló</b>t tartunk fenn vendégeink számára, valamint házunk megfelel a FATOSZ által elvárt minősítési követelményeknek.</p>
    {/* <!-- TODO Consider adding zoom-in / zoom-out feature for the certificates--> */}
    <div id="certificates">
    </div>
  </section>
)

export default Sunflower