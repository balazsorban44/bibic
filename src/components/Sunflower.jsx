import React from 'react'
import Paragraphs from './shared/Paragraphs'
import Gallery from './shared/Gallery'

const Sunflower = () =>
  <section id="napraforgo">
    <Paragraphs path="tanusitvanyok"/>
    <div className="certificates">
      <Gallery path="tanusitvanyok"/>
    </div>
  </section>


export default Sunflower