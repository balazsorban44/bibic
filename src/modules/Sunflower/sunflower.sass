@import 'ui/_vars'

#napraforgo
  background:
    color: #614902
    size: cover
  position: relative
  overflow: hidden
  
  .gallery
    margin: 2em 0
    display: flex
    flex-wrap: wrap
    justify-content: center
    img
      background-color: transparentize($accent-color-dark, .1)
      color: $light
      font-size: .6em
      text-align: center
      align-items: center
      margin: .25em
      min-width: 160px
      border: 4px solid $accent-color-3
      height: 200px


@media (min-width: 480px)
  #napraforgo
    p
      margin: auto
      max-width: 90%

@media (min-width: 760px)
  #napraforgo
    background-color: #cc9900

@media (min-aspect-ratio: 3/2)
  #napraforgo
    background: linear-gradient(0deg, transparent, transparentize(#000000, .2)), url("./sunflower.jpg")
    background-size: cover
    p
      max-width: 720px
    display: flex
    align-items: flex-end
    flex-direction: column
    background-color: transparent
    .gallery
      justify-content: flex-end
      img
        transition: .2s
        margin-right: 1em
        &:hover
          transform: scale(1.05)