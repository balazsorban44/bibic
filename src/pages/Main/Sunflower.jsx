import React from 'react'
import Paragraphs from 'components/Paragraphs'
import Gallery from 'components/Gallery'
import CERTIFICATES_BG from "assets/images/bg/intro.jpg"

const Sunflower = () =>
  <section id="napraforgo">
    <Paragraphs section="certificates"/>
    <Gallery section="certificates"/>
    <div className="cert-bg">
      <img
        alt="certificates background"
        src={CERTIFICATES_BG}
      />
    </div>
  </section>


export default Sunflower