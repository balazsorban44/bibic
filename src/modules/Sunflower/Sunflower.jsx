import React from "react"
import Carousel from "components/Carousel"
import Paragraphs from "components/Paragraphs"
import CERTIFICATES_BG from "assets/images/bg/intro.jpg"
import {GalleryItem} from "components/Gallery"
import useMedia from "ui/utils/useMedia"

const Sunflower = () => {
  const matches = useMedia("(min-aspect-ratio: 3/2)")

  return (
    <section id="napraforgo">
      <Paragraphs paragraphProps={{align: matches ? "right" : "left"}} type="tanusitvanyok"/>
      <Carousel
        className="gallery"
        item={GalleryItem}
        slider={false}
        type="tanusitvanyok"
      />
      <div className="cert-bg">
        <img
          alt="certificates background"
          src={CERTIFICATES_BG}
        />
      </div>
    </section>
  )
}


export default Sunflower