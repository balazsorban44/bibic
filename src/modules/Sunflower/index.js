import Sunflower from "./Sunflower"

export default Sunflower