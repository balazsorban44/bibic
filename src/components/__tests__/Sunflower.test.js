import Sunflower from "../Sunflower"

describe("Sunflower component", () => {
  const wrapper = shallow(<Sunflower/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

})