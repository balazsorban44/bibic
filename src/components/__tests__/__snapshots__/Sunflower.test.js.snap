// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sunflower component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sunflower />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Component
          path="tanusitvanyok"
        />,
        <Component
          path="tanusitvanyok"
        />,
        <div
          className="cert-bg"
        >
          <img
            alt="certificates background"
            src="intro.jpg"
          />
        </div>,
      ],
      "id": "napraforgo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "tanusitvanyok",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "tanusitvanyok",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="certificates background"
            src="intro.jpg"
          />,
          "className": "cert-bg",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "certificates background",
            "src": "intro.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Component
            path="tanusitvanyok"
          />,
          <Component
            path="tanusitvanyok"
          />,
          <div
            className="cert-bg"
          >
            <img
              alt="certificates background"
              src="intro.jpg"
            />
          </div>,
        ],
        "id": "napraforgo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "tanusitvanyok",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "tanusitvanyok",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="certificates background"
              src="intro.jpg"
            />,
            "className": "cert-bg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "certificates background",
              "src": "intro.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
